Include <sys/socket.h> before including <net/if.h> for compatibility
with OS X 10.8 and earlier.

https://gitlab.com/libvirt/libvirt/-/merge_requests/13
--- configure.ac.orig
+++ configure.ac
@@ -381,7 +381,6 @@ AC_CHECK_HEADERS([\
   linux/magic.h \
   mntent.h \
   net/ethernet.h \
-  net/if.h \
   pty.h \
   pwd.h \
   stdarg.h \
@@ -393,6 +392,7 @@ AC_CHECK_HEADERS([\
   sys/ucred.h \
   xlocale.h \
   ])
+AC_CHECK_HEADERS([net/if.h], [], [], [[#include <sys/socket.h>]])
 dnl Check whether endian provides handy macros.
 AC_CHECK_DECLS([htole64], [], [], [[#include <endian.h>]])
 AC_CHECK_FUNCS([stat stat64 __xstat __xstat64 lstat lstat64 __lxstat __lxstat64])
--- src/util/virnetdev.h.orig
+++ src/util/virnetdev.h
@@ -18,10 +18,6 @@
 
 #pragma once
 
-#ifdef HAVE_NET_IF_H
-# include <net/if.h>
-#endif
-
 #include "virbitmap.h"
 #include "virsocketaddr.h"
 #include "virmacaddr.h"
@@ -29,6 +25,10 @@
 #include "virnetdevvlan.h"
 #include "virenum.h"
 
+#ifdef HAVE_NET_IF_H
+# include <net/if.h>
+#endif
+
 #ifdef HAVE_STRUCT_IFREQ
 typedef struct ifreq virIfreq;
 #else
